tests/test_regression/test_losses.py

```diff
@@ -62,16 +72,26 @@ class SLCWALossCalculator(LossCalculator):
 
     num_negatives: int
 
+    weighted: bool = False
+
     # docstr-coverage: inherited
     def __call__(self, instance: Loss, generator: torch.Generator) -> torch.Tensor:
         positive_scores = torch.rand(self.batch_size, 1, generator=generator)
         negative_scores = torch.rand(self.batch_size, self.num_negatives, generator=generator)
+        if self.weighted:
+            pos_weights = torch.rand(self.batch_size, 1, generator=generator)
```

Comment:
@mberr take a look at how I incorporated the weights, and if you like this, we can add regression tests for weighted losses too just by rerunning `python tests/test_regression/test_losses.py`